compiler/kernel/expr.go

```diff
@@ -73,6 +73,10 @@ func (b *Builder) compileExpr(e dag.Expr) (expr.Evaluator, error) {
 		return b.compileConditional(*e)
 	case *dag.Call:
 		return b.compileCall(*e)
+	case *dag.IndexExpr:
+		return b.compileIndexExpr(e)
+	case *dag.SliceExpr:
+		return b.compileSlice(e)
```

Comment:
Nit: Call this compileSliceExpression so it matches the type as with the other Builder.compile* methods.